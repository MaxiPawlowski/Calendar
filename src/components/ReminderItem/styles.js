import { createStyles } from '@material-ui/core';

export default () =>
  createStyles({
    container: {
      display: 'flex',
      borderRadius: '4px',
      justifyContent: 'space-between',
      padding: '10px',
      margin: '10px 0'
    },
    icon: {
      cursor: 'pointer',
    },
    iconsContainer: {
      display: 'flex',
      marginLeft: '10px',
      justifyContent: 'space-between',
    },
    typography: {
      display: 'block',
      verticalAlign: 'middle',
      lineHeight: 'normal',
      margin: '5px 0',
    },
  });
