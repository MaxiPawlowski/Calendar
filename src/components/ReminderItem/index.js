import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import DeleteForeverOutlinedIcon from '@material-ui/icons/DeleteForeverOutlined';
import EditOutlinedIcon from '@material-ui/icons/EditOutlined';
import { parseTime } from 'utils/date'
import styles from "./styles";

const hexToRGB = (hex, alpha = 1) => 
  `${parseInt(hex.slice(1, 3), 16)}, ${parseInt(hex.slice(3, 5), 16)}, ${parseInt(hex.slice(5, 7), 16)}, ${alpha}`;


const ReminderItem = ({ classes, reminder, deleteReminder, editReminder }) => (
  <div
    className={classes.container}
    style={{
        border: `1px solid ${reminder.color}`, 
        background: `rgba(${hexToRGB(reminder.color, 0.4)})`, 
      }}
  >
    <div>
      <Typography className={classes.typography}>
        {`${parseTime(reminder.date)} - ${reminder.title}`}
      </Typography>

      {reminder.city && (
      <Typography className={classes.typography}>
        {`${reminder.city.description}`}
        {reminder.weather && `. ${reminder.weather.main} (${reminder.weather.description})`}
      </Typography>
        )}
    </div>
    <div style={reminder.city ? {flexDirection:'column'}:{}} className={classes.iconsContainer}>
      <EditOutlinedIcon className={classes.icon} onClick={() => editReminder(reminder)} />
      <DeleteForeverOutlinedIcon className={classes.icon} onClick={() => deleteReminder(reminder)} />
    </div>
  </div>
);


ReminderItem.propTypes = {
  classes: PropTypes.shape({
    container: PropTypes.string.isRequired,
    typography: PropTypes.string.isRequired,
    iconsContainer: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
  }).isRequired,

  reminder: PropTypes.shape({
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date).isRequired,
    id: PropTypes.number.isRequired,
      city: PropTypes.oneOfType([
        PropTypes.string.isRequired,
        PropTypes.shape({
          description: PropTypes.string.isRequired,
          location: PropTypes.shape({
            lat: PropTypes.number.isRequired,
            lng: PropTypes.number.isRequired,
          }).isRequired
        }),
      ]),
    }).isRequired,

  deleteReminder: PropTypes.func.isRequired,
  editReminder: PropTypes.func.isRequired,
};

export default withStyles(styles)(ReminderItem);
