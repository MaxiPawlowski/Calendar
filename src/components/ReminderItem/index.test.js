import React from 'react';
import { shallow } from 'enzyme';
import ReminderItem from "./index";

describe('ReminderItem', () => {
  let wrapper;

  describe('when rendered', () => {
    beforeAll(() => {
      wrapper = shallow(
        <ReminderItem
          editReminder={()=>{}}
          deleteReminder={()=>{}}
          reminder={{
            title: "test",
            color: "#246751",
            date: new Date(Date.now()),
            city: "",
            id: 1571083793866
          }}
        />
      );
    });

    it('should render correctly', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });
  });
});
