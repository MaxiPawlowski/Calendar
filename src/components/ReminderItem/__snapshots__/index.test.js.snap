// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReminderItem when rendered should render correctly 1`] = `
<div
  className="ReminderItem-container-1"
  style={
    Object {
      "background": "rgba(36, 103, 81, 0.4)",
      "border": "1px solid #246751",
    }
  }
>
  <div>
    <WithStyles(ForwardRef(Typography))
      className="ReminderItem-typography-4"
    >
      09:51 - test
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div
    className="ReminderItem-iconsContainer-3"
    style={Object {}}
  >
    <pure(EditOutlinedIcon)
      className="ReminderItem-icon-2"
      onClick={[Function]}
    />
    <pure(DeleteForeverOutlinedIcon)
      className="ReminderItem-icon-2"
      onClick={[Function]}
    />
  </div>
</div>
`;
